### ISSUE TITLE
Update to createKnativeService value in app-deploy.yaml does not apply to appsodyApplication CR

### ISSUE THREAD

**Issue by tam512** [kabanero-io/kabanero-pipelines#166]
- Use Kabanero 0.5.0-rc2 to test Java microprofile app with createKnativeService enabled
- Enabled `createKnativeService` to `true` in app-deploy.yaml to trigger a java-microprofile-build-push-deploy-pipeline-run, after the build is done running, I do not see Knative resources being created in `oc get all` output.
- Below is the settings in app-deploy.yaml file
```
spec:
  createKnativeService: true
  expose: true
```
- After the pipleline done building, `oc get appsodyApplication javamp -o yaml` output shows `createKnativeService: false`

```
# oc get appsodyApplication javamp -o yaml
apiVersion: appsody.dev/v1beta1
kind: AppsodyApplication
metadata:
  annotations:
    architecture: x86_64
    ............................
    .................
spec:
  applicationImage: image-registry.openshift-image-registry.svc:5000/svt-project3/javamp:94af9a3c168e0d42e8a1e7920aa8207cdb63ff94
  createKnativeService: false
  expose: true
  livenessProbe:
    failureThreshold: 12
    httpGet:
      path: /health/live
      port: 9080
    initialDelaySeconds: 5
    periodSeconds: 2
  pullPolicy: IfNotPresent
  readinessProbe:
    failureThreshold: 12
    httpGet:
      path: /health/ready
      port: 9080
    initialDelaySeconds: 5
    periodSeconds: 2
    timeoutSeconds: 1
  resourceConstraints: {}
  service:
    annotations:
      prometheus.io/scrape: "true"
    port: 9080
    type: NodePort
  stack: java-microprofile
  version: 1.0.0
status:
  conditions:
  - lastUpdateTime: "2020-01-22T14:39:19Z"
    status: "True"
    type: DependenciesSatisfied
  - lastUpdateTime: "2020-01-22T14:39:19Z"
    status: "True"
    type: Reconciled
```

**Comment by tam512** [comment]
Below is @navidst 's comment after helping debug the problem: 

Based on the investigation, it looks like the AppsodyApplication CR deployed, has createKnativeService: false even though app-deploy.yaml in the repo has createKnativeService: true. Which means, the Appsody operator deploys using regular Deployment.

We verified that Appsody Operator is not resetting createKnativeService from true to false, by manually changing createKnativeService in the CR.

**Comment by marikaj123** [comment]
Appsody team need to provide the fix and then Kabanero can include it.

**Comment by marikaj123** [comment]
@mtamboli will be opening an issue with Appsody.

**Comment by tam512** [comment]
I opened appsody CLI defect https://github.com/appsody/appsody/issues/865  to fix this problem.
Keep this defect open to update all the pipelines to pickup the new appsody image with the fix

**Comment by marikaj123** [comment]
Appsody-cli working to fix by today.

**Comment by kylegc** [comment]
The fix is part of image `appsody/appsody-buildah:0.5.6-buildah1.9.0`
The pipelines should be updated to use this image.

**Comment by marikaj123** [comment]
Pipeline team is including the fix.

**Comment by kvijai82** [comment]
Fixed is merged to pushed to collections in PR https://github.com/kabanero-io/collections/pull/256.  SVT, please verify issue is resolved in the latest collections RC.

**Comment by marikaj123** [comment]
@mtamboli - Hi Monica, Please close after the issue is verified successfully.  Thank you.

**Comment by tam512** [comment]
Verify that this problem is fixed in Kabanero 0.5.0-rc.3 and collection 0.5.0-rc.4.
The following knative resources were create after deploy with `createKnativeService: true`
```
NAME                                 URL                                                     LATESTCREATED   LATESTREADY    READY   REASON
service.serving.knative.dev/javamp   http://javamp.svt-project1.apps.toast.os.fyre.ibm.com   javamp-jddk9    javamp-jddk9   True

NAME                                       LATESTCREATED   LATESTREADY    READY   REASON
configuration.serving.knative.dev/javamp   javamp-jddk9    javamp-jddk9   True

NAME                                        CONFIG NAME   K8S SERVICE NAME   GENERATION   READY   REASON
revision.serving.knative.dev/javamp-jddk9   javamp        javamp-jddk9       1            True

NAME                               URL                                                     READY   REASON
route.serving.knative.dev/javamp   http://javamp.svt-project1.apps.toast.os.fyre.ibm.com   True
```
